test/src/test/java/hudson/tasks/ArtifactArchiverTest.java

```diff
@@ -131,6 +131,20 @@ public void testAllowEmptyArchive() throws Exception {
         assertFalse(project.getBuildByNumber(1).getHasArtifacts());
     }
 
+    @Test
+    @Issue("JENKINS-51913")
+    public void testFileMaskNoMatchesFoundException() throws Exception {
+        FreeStyleProject project = j.createFreeStyleProject();
+        String pattern = "dir/**";
+        ArtifactArchiver aa = new ArtifactArchiver(pattern);
+        aa.setAllowEmptyArchive(true);
+        project.getPublishersList().replaceBy(Collections.singleton(aa));
+        FreeStyleBuild build = j.buildAndAssertSuccess(project);
+        assertFalse(project.getBuildByNumber(1).getHasArtifacts());
+        j.assertLogContains("No artifacts found that match the file pattern \"" + pattern + "\"", build);
+        assertThat("No stacktrace shown", build.getLog(31), Matchers.iterableWithSize(lessThan(30)));
```

Comment:
Or more directly could do something along the lines of

```java
assertThat("No stacktrace shown", build.getLog(), not(containsString("\tat ")));
```

Reply:
But then that would be inconsistent with other code in this file.